src/Serilog/Events/LogEvent.cs

```diff
@@ -192,3 +194,18 @@ internal LogEvent Copy()
             properties);
     }
 }
+
+internal static class LogEventExtensions
```

Comment:
@nblumhardt For now it's `internal` but it may be changed to public for use in dependent packages. For example, in serilog-extensions-logging, see `EnrichAndCreateScopeItem`: 

```
var property = propertyFactory.CreateProperty(key, value, destructureObject);
logEvent.AddPropertyIfAbsent(property);
```